doc/12-upgrading.md

```diff
@@ -3,14 +3,4 @@
 Once you upgraded the bundle to a newer version, you first have to 
 be sure your current schema is up to date and upgrade it if necessary.
 
-The following command does all of this for you:
-
-```bash
-# symfony < 3.4
-app/console audit:schema:update --force
-```
-
-```bash
-# symfony >= 3.4
-bin/console audit:schema:update --force
-```
```

Comment:
This should not be removed because, you have to run this command whenever you upgrade to be sure that every audit table structure is up to date, no matter if you are using a single ou dual database setup.

Reply:
The notice about the lock package should be also added here (at the top of this section)

Reply:
Sorry, not sure about this. The upgrading steps also depend on which process you're using. If you use migrations you want to run `app/console doctrine:migrations:diff` and `app/console doctrine:migrations:migrate` to update. If you just use the `doctrine:schema:update` command then it's fine to use this for upgrading the tables. You should only require `app/console audit:schema:update` command if you're on a multi-db setup. So in this section it would describe exactly the same as in `22-audit-schema.md` which is why i linked to it. Or am i missing something?

Reply:
@n3o77 In fact, the upgrade process has to be done with `audit:schema:update` command, **no matter if you are in a single or multi db setup** because the schema listener is only able to create new audit tables, it is not able to update existing audit tables. This is by design and won't change for several reasons:
- the upgrade process implies altering audit tables to add/remove/change columns and indices 
=> this can be very resource intensive if existing tables are big, that's the main reason why it is a separate dedicated process: the `audit:schema:update` command
- the schema listener is only "run" when there is a schema change on a per table basis so even if we assume it is able to do the update stuff, it won't be able to update other audit tables than the one related to the table originating the event

Then, the `audit:schema:update` commands is able to do the stuff or print the relevant SQL queries so you can copy and paste them in a migration.
But, as migrations can only use one connection at a time, this is not very convenient in a multi db setup. 

That are the reason why I think we have to keep the upgrading section of the documentation as it is now but with the notice added at the top ;)

Reply:
Thanks for clarifying. Actually that's an problem as I couldn't use the `audit:schema:update` command in a live system as everything is done via migrations. Otherwise you can't downgrade properly. In this case I also wouldn't agree on moving the `symfony/lock` to dev as you need it in production to update the audit tables.

Reply:
Good catch for the lock requirement! You're right, it has to be move to regular `require` section.
Regarding the downgrading subject, I still don't have a better proposal :/ 
By the way, when I need/plan to alter the audit tables for a future version (happened only twice I think), I try to do it in a way that reverting the bundle to prior version is not a problem with the new structure but we never know... it still can happen one day.

Reply:
I think it's quite tricky, unfortunately i'm not deep enough in this bundle to help you there. We had a multi-tenant project last year and based the migration / updating stuff of the multiple databases loosely based on this demo: https://github.com/uirapuru/multidb

Maybe something like that could be used here too but not sure.

Reply:
Thanks for the heads up

Reply:
I think we only miss the notice in the upgrading doc to merge this PR now ;)